src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -408,9 +408,14 @@ private synchronized int execMatlabCommand(Run<?, ?> build, FilePath workspace,
             TaskListener listener, boolean isLinuxLauncher)
             throws IOException, InterruptedException {
         setEnv(build.getEnvironment(listener));
-        //Copy MATLAB scratch file into the workspace
-        copyMatlabScratchFileInWorkspace(MATLAB_RUNNER_RESOURCE, MATLAB_RUNNER_TARGET_FILE,
-                workspace, getClass().getClassLoader());
+        final String testRunMode = this.getTestRunTypeList().getDescriptor().getDisplayName();
+        
+        // Copy MATLAB scratch file into the workspace only if Automatic option is selected.
+        if (!testRunMode.equalsIgnoreCase(
```

Comment:
Curious why we dont do this operation only in automatic mode vs when it is not custom mode. AUtomatic seems more direct and makes less assumptions.

Reply:
In general it seems like this code could be refactored to avoid the code branches based on these two modes. Somethign to consider as we break out the build into two steps.